share/wake/lib/core/json.wake

```diff
@@ -538,3 +538,162 @@ export def lastValueInList (values: List JValue): Result JValue Error =
             | flatten
             | deduplicateJObjectKeys lastValueInList
             | rmap JObject
+
+export def jString: JValue => List String =
+    def view j =
+        require (JString s) = j else Nil
+        s, Nil
+    view
```

Comment:
Why not just ...?
```
export def jString: JValue => List String = match _
  JString s = s, Nil
  _ = Nil
````

Reply:
(also, you don't needs the `()`s on your requires)

Reply:
Yeah I'll rewrite all these. I edited this all from the previous code where there was both a view and an edit and so the quickest thing to do was to 1) duplicate the functions 2) delete the now unused parts 3) rename them. My objective here was to put the minimum amount of work in to get feedback but looks like we're happy with this now and we can move forward! I'll start getting this into a more polished state now.

Also I didn't think about that on the require for some reason! I've been writing `require Pass x = ...` and same for `Some` but for everything else I used '(Foo x y z)` for some reason. Thanks!

Reply:
I understand why you did it. You wrote: `x (y z)` because you didn't want `x y z`. It just happens that when `x` is `def` or `require` the parser has a special behaviour. Unfortunately, I wasn't consistent with `match`. It does NOT have the special handling that `def` and `require` get. I'm not sure how many places take advantage of the multi-match syntax of `match x y z`...